src/client/src/rt-system/__tests__/ServiceStub.test.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  AutobahnSessionProxy,
+  ConnectionEvent,
+  ConnectionEventType,
+  ConnectionType,
+  ConnectionOpenEvent,
+} from 'rt-system'
+import { MockScheduler } from 'rt-testing'
+import { ServiceStub } from 'rt-system'
+import { of } from 'rxjs'
+
+const createMockConnection: (session: AutobahnSessionProxy) => ConnectionOpenEvent = (
+  session: AutobahnSessionProxy,
+) => ({
+  type: ConnectionEventType.CONNECTED,
+  session,
+  url: 'FAKE',
+  transportType: ConnectionType.LongPolling,
+})
+
+describe('ServiceStub', () => {
+  const payload = 'payload'
+  const replyTo = 'responseTopic'
+  const Username = 'USER'
+  const procedure = 'procedure'
+  const result = 'result'
+
+  describe('Request Response', () => {
+    it('invokes the a remote procedure with the correct payload', () => {
+      const session = new MockSession()
+      const stub = new ServiceStub(Username, of(createMockConnection(session)))
+      const rpc = stub.requestResponse<string, string>(procedure, payload, replyTo).subscribe()
+      expect(session.call).lastCalledWith(procedure, [{ payload, replyTo, Username }])
```

Comment:
Correct me if I am wrong, but I think it is redundant to test whether it returns the correct payload as the responsibility for a correct payload lies with the back-end side. As such, I will assume that it shouldn't be tested.

Reply:
We are testing that we call the session object with the correct payload. We are also testing that the correct value is returned when from the session object. The session object is mocked.